### ISSUE TITLE
Possible improvement for method 'html2unicode'

### ISSUE THREAD

**Issue by OshinoMoumou** [klahnakoski/esShardBalancer#3]
Greetings!
I noticed that method html2unicode referred a StackOverflow post, while [a recent comment](https://stackoverflow.com/questions/57708/convert-xml-html-entities-into-unicode-string-in-python#comment67119532_12614706) to it mentioned that this raise UnicodeDecodeError with utf-8 strings. You must either decode('utf-8') it first or use xml.sax.saxutils.unescape..
I'm trying to learn the usefulness of such small updates on StackOverflow. Would this comment help improve your code? I understand that such defect might not happen in real life situation. In that case, do you think this comment can help prevent future bugs (for example, when the code were reused somewhere else)?
I'll really appreciate it if you could kindly give me some feedback or suggestions. Thank you very much for your time.
Have a nice day!

**Comment by klahnakoski** [comment]
No, performing this would not improve the code.  it may make it worse.

Each function should do something simple or unsurprising; this rarely happens, but that's what should happen.  In the case of `html2unicode` it takes 7bit HTML and converts it to Unicode.  If someone feeds it Unicode, then it could produce something unexpected, and that's OK.  There are many bad things that can go into the function; numbers, objects, hexdecimal byte arrays, etc.  They will all result in something bad.  That said, the `UnicodeDecodeError` exception is a **good** output: It identifies the problem and reports it to the caller.

`html2unicode` exists as a facade over the `HTMLParser.unescape()` for two reasons: 1) to be easier for me to remember, and 2) to isolate my code from irregular behaviour found in library functions. I doubt `unescape` actually works on all legitimate HTML; it is rare a library function is compete and works on the corner cases, so I use a facade to handle those cases (as you can see, there are none I found yet).  Here is a [URI converter facade](https://github.com/mozilla-frontend-infra/firefox-health-dashboard/blob/b0dabd411bf86b9f7ed72798dd1b1be8d806f94a/src/vendor/requests.js#L48) that fixes the output; converting "%20" to "+".

Maybe we can add "type-checking" code to html2unicode:  Maybe we can verify the input is a 7bit byte array, but that check is expensive in Python.  Maybe we can add a try/catch block to report `UnicodeDecodeError`, but the stack trace already has enough information.

The proper solution is to add testing.  Your code should be tested.  At the very least manual testing, but automated tests will help you immensely.  You want to write tests that verify the action your code is taking.  Your tests will cover `html2unicode`, and you can verify the function works on the type of situations your code needs it for.
